matrix_sdk_base/src/models/room.rs

```diff
@@ -516,20 +679,9 @@ impl Room {
     /// # Arguments
     ///
     /// * `event` - The event of the room.
-    pub fn receive_state_event(&mut self, event: &StateEvent) -> bool {
-        match event {
-            // update to the current members of the room
-            StateEvent::RoomMember(member) => self.handle_membership(member),
-            // finds all events related to the name of the room for later use
-            StateEvent::RoomName(name) => self.handle_room_name(name),
-            StateEvent::RoomCanonicalAlias(c_alias) => self.handle_canonical(c_alias),
-            StateEvent::RoomAliases(alias) => self.handle_room_aliases(alias),
-            // power levels of the room members
-            StateEvent::RoomPowerLevels(power) => self.handle_power_level(power),
-            StateEvent::RoomTombstone(tomb) => self.handle_tombstone(tomb),
-            StateEvent::RoomEncryption(encrypt) => self.handle_encryption_event(encrypt),
-            _ => false,
-        }
+    pub fn receive_state_event(&mut self, event: &AnyStateEventStub, room_id: &RoomId) -> bool {
+        // forward to the room event handlers.
+        self.receive_timeline_event(&AnyRoomEventStub::StateEvent(event.clone()), room_id)
```

Comment:
It seems to me that because the types from ruma are more accurate now we could remove some of these methods and just have a `receive_sync_event`?

Reply:
Since we have the `room_id` with the sync response we could convert all the `receive_*_event` and `on_*_event` methods to `receive_*_event(&self, event: AnyRoomEvent)` by converting the `AnyRoomEventStub` to a `AnyRoomEvent`. It may remove the need for conversions.